newIDE/app/src/ProjectManager/index.js

```diff
@@ -714,6 +533,24 @@ export default class ProjectManager extends React.Component<Props, State> {
     this._onProjectItemModified();
   };
 
+  _onEditEventsFunctionExtension = (
+    extensionShortHeadersByName: { [string]: ExtensionShortHeader },
+    eventsFunctionsExtension: gdEventsFunctionsExtension,
+    name: string
+  ) => {
+    const originName = eventsFunctionsExtension.getOriginName();
+    if (originName !== 'gdevelop-extension-store') {
+      this.props.onOpenEventsFunctionsExtension(name);
+      return;
+    }
+    const originIdentifier = eventsFunctionsExtension.getOriginIdentifier();
+    const extensionShortHeader = extensionShortHeadersByName[originIdentifier];
```

Comment:
Is this possible to be undefined? Will this create an issue?

Reply:
Worth adding a comment here and above/below detailing the logic :
"If the extension is not coming from the store, or is but can't be found, edit it directly:"
"Otherwise open its details first"